.github/workflows/main.yml

```diff
@@ -0,0 +1,348 @@
+name: CI
+
+on:
+  # TODO: add nightly schedule.
+  pull_request:
+    branches:
+    - master
+  push:
+    branches:
+    - master
+  release:
+    types: [published]
+
+jobs:
+  check_core:
+    name: Check Core
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7']
```

Comment:
I know you just moved things from files, but maybe we should make the tests run on 3.6-8? As these are the 3 major Python versions we support.

Reply:
I'm not sure if we should support 3.8 yet. Last time I tried to use 3.8 I ran into a butch of issues with dependencies not supporting it yet. That was a couple months ago so maybe those issues are fixed by now. In any case it would be worth trying but should probably hold off until we get the release process completely ironed out.

Reply:
@bryant1410, did you try it with 3.8? Does it work?

Reply:
I use a project with Python 3.8 with AllenNLP from master